optimum/neuron/distributed/utils.py

```diff
@@ -23,19 +23,58 @@
 
 if is_neuronx_distributed_available():
     from neuronx_distributed.parallel_layers import layers
-    from neuronx_distributed.parallel_layers.parallel_state import (
-        get_tensor_model_parallel_rank,
-    )
+    from neuronx_distributed.parallel_layers.parallel_state import get_tensor_model_parallel_rank
 
 TENSOR_PARALLEL_SHARDS_DIR_NAME = "tensor_parallel_shards"
 
 
+def embedding_to_parallel_embedding(
+    embedding_layer: "torch.nn.Embedding",
+    lm_head_layer: Optional["torch.nn.Linear"] = None,
+    orig_to_parallel: Optional[Dict[int, "torch.nn.Parameter"]] = None,
+) -> Union["layers.ParallelEmbedding", Tuple["layers.ParallelEmbedding", "layers.ColumnParallelLinear"]]:
+    parallel_embedding_layer = layers.ParallelEmbedding(
+        embedding_layer.num_embeddings,
+        embedding_layer.embedding_dim,
+        dtype=embedding_layer.weight.dtype,
+    )
+
+    tp_rank = get_tensor_model_parallel_rank()
+    row_size, _ = parallel_embedding_layer.weight.shape
+
+    is_tied = False
+    if lm_head_layer is not None:
+        is_tied = id(embedding_layer.weight.data) == id(lm_head_layer.weight.data)
```

Comment:
If this is xla_tensor, then the data pointer may not be same. This would mean the weight has to be a CPU tensor first